pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -50,12 +50,16 @@ func (r *resourceAllocationScorer) score(
 		return 0, framework.NewStatus(framework.Error, "resources not found")
 	}
 
-	requested := make([]int64, 0, len(r.resources))
-	allocatable := make([]int64, 0, len(r.resources))
+	requested := make([]int64, len(r.resources))
+	allocatable := make([]int64, len(r.resources))
 	for i := range r.resources {
 		alloc, req := r.calculateResourceAllocatableRequest(nodeInfo, pod, v1.ResourceName(r.resources[i].Name))
-		allocatable = append(allocatable, alloc)
-		requested = append(requested, req)
+		// Only fill the extended resource entry when it's non-zero.
+		if alloc == 0 {
+			continue
+		}
```

Comment:
leave this and revert the rest. It's better if we leave the list shorter and it's ok to "waste" a bit of memory.

Reply:
There may be a bug if the node doesn't have the resource we defined in ScoringStrategy(see test "if the node doesn't have a resource"), then we'll have an unmatched index in requested with resources. So we should append the value here whatever the value.

Reply:
@kerthcet talk about this earlier.

Reply:
Yes. We used to use map which can avoid this.